pkg/scuzz/sg_linux.go

```diff
@@ -227,7 +249,22 @@ func (s *SGDisk) UnlockRequest(password string, timeout uint, master bool) Reque
 	return &sgRequest{packet: p}
 }
 
-func (s *SGDisk) Operate(r Request) error {
+func (s *SGDisk) IdentifyRequest(timeout uint) Request {
+	p := s.newPacket(identify, from, 0, timeout)
+	p.genCommandDataBlock()
+	return &sgRequest{packet: p}
+}
+
+func (s *SGDisk) Operate(r Request) (Response, error) {
 	_, _, err := unix.Syscall(unix.SYS_IOCTL, uintptr(s.f.Fd()), r.Cmd(), r.Packet())
-	return err
+	if err != 0 {
+		return nil, err
+	}
+	var resp = &sgResponse{}
+	if r.(*sgRequest).packet.status[0] != 0 {
```

Comment:
You don't need the cast

Reply:
So is Status == error? Whats in this Byte slice?

Reply:
I'm happy to see the code which will work without the cast, but for now, it's needed. r is a Request. status is whatever chunk of bytes the drive returns for the request. It's essentially the rpc response.

Reply:
btw, the slice is defined in the ata13 standard which I am still striving to understand.

Reply:
O I misread r as resp.